python-main.js

```diff
@@ -1506,19 +1498,13 @@ function web_editor(config) {
 
         var p = Promise.resolve();
 
-        if (usePartialFlashing) {
-            if (window.dapwrapper) {
-                console.log("Disconnecting: Using Quick Flash");
-                p = p.then(function() { window.dapwrapper.daplink.stopSerialRead() } )
-                    .then(function() { window.dapwrapper.disconnectAsync() } );
-            }
+        if (usePartialFlashing && window.dapwrapper) {
+            console.log('Disconnecting: Using Quick Flash');
+            p = p.then(function() { return window.dapwrapper.disconnectAsync() });
```

Comment:
@microbit-sam I think these two were meant to return the disconnect promise?

I assume this was originally an arrow function `.then(() => window.dapwrapper.disconnectAsync());` that returned the promise from disconnectAsync(), but then when converting to ES5 the return was accidentally left out?

On the other hand stopSerialRead() does **not** return a promise, so the original line removed in this PR (`p.then(function() { window.dapwrapper.daplink.stopSerialRead() } )`) is correct in not using `return` in the function.

Reply:
Yep, looks right!